### ISSUE TITLE
local state mismatches API state due to too aggressive redux-persist caching

### ISSUE THREAD

**Issue by andreasn** [weldr/welder-web#163]
I think I have something weird going on with the cache in Firefox, because I get a lot of Error messages when I try to click on a recipe in the test instance.
However, when I tried this in Chrome I got the empty page, so I suspect Firefox have cached old recipes for some reason. Shift+reload doesn't quite do the trick, so there must be something else going on.

**Comment by andreasn** [comment]
@martinpitt

**Comment by martinpitt** [comment]
This is not a browser cache issue, but the problem is rooted in the usage of redux-persist. This writes the entire state, including recipes and pending changes, into the Browser's sessionstore, so it survives page reloads, restarts, etc. Once the backend state changes (because some other person edits it), the local redux store and the "real" state from the backend get out of sync, but apparently the redux store always wins.

You can also see this by stopping the `api` container/process, then most of the UI will still happily work.

I'm not sure what the solution is, though. @jkozol, any idea?

**Comment by martinpitt** [comment]
BTW, the only way how to clean this up that I found is this:
```
$ sqlite3 ~/.mozilla/firefox/*default*/webappsstore.sqlite
sqlite> delete from webappsstore2 where key == 'reduxPersist:recipes';
```

**Comment by martinpitt** [comment]
@jkozol, @atodorov, @jgiardino : Can one of you please rename this to "local state mismatches API state due to too aggressive redux.persist caching" or something such? Right now the issue title is not very useful. I'm not a project member, so I can't change it. (BTW, maybe I can become one?)

**Comment by jgiardino** [comment]
Looks like this will be closed with #245
